lib/Cake/Model/Datasource/DboSource.php

```diff
@@ -2550,14 +2550,17 @@ public function fields(Model $Model, $alias = null, $fields = array(), $quote =
 			}
 			return $fields;
 		}
-		$count = count($fields);
 
+		$count = count($fields);
 		if ($count >= 1 && !in_array($fields[0], array('*', 'COUNT(*)'))) {
 			for ($i = 0; $i < $count; $i++) {
 				if (is_string($fields[$i]) && in_array($fields[$i], $virtual)) {
 					unset($fields[$i]);
 					continue;
 				}
+
+				$fields[$i] = preg_replace('/\t/', '', $fields[$i]);
+				$fields[$i] = str_replace(array("\r", "\n"), '', $fields[$i]);
```

Comment:
I have made a new commit removing the line 2562. so it just will remove the break lines then the query will stay in the same line and this regex https://github.com/CakeDC/cakephp/blob/hot-fix/distinct-with-break-line/lib/Cake/Model/Datasource/DboSource.php#L2567 won't match with with the query with break line.